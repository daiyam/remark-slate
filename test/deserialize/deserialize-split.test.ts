import unified from 'unified';
import markdown from 'remark-parse';
import plugin from '../../src';

/* it('Deserialize a bold paragraph with a continuous split', () => {
  unified()
    .use(markdown)
    .use(plugin)
    .process('**bold para****graph**', (_, file) => {
      //@ts-ignore ts doesn't know about file.result
      const res = file.result;
      expect(res).toMatchSnapshot();
    })
}); */

it('Deserialize a bold paragraph splitted by a space', () => {
  unified()
    .use(markdown)
    .use(plugin)
    .process('**bold** **paragraph**', (_, file) => {
      //@ts-ignore ts doesn't know about file.result
      const res = file.result;
      expect(res).toMatchSnapshot();
    })
});
