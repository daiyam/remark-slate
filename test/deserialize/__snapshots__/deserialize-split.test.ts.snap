// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deserialize a bold paragraph splitted by a space 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "bold": true,
        "text": "bold",
      },
      Object {
        "text": " ",
      },
      Object {
        "bold": true,
        "text": "paragraph",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`Deserialize a bold paragraph with a continuous split 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "bold": true,
        "text": "bold para",
      },
      Object {
        "bold": true,
        "text": "graph",
      },
    ],
    "type": "paragraph",
  },
]
`;
